packages/build-info/src/build-systems/nrwl.ts

```diff
@@ -68,6 +79,7 @@ export class Nx extends BaseBuildTool {
               packages,
               rootDir: this.project.jsWorkspaceRoot,
             }
+            this.project.events.emit('detectWorkspaces', this.project.workspace)
```

Comment:
Isn't this handled by project.ts line 195?

Reply:
nope (the general workspace detection runs previously before the framework detection (as the framework detection needs it to know all packages from a workspace root to take a look at)

In case of Nx integrated there is only one package.json in the root (and it's not a pnpm, yarn or npm workspace) so we need to leverage nx to get the packages and then emit the update. (so detect workspaces is emitted twice in this case).

There is a test exactly testing that (the added one in `project.test.ts`)